src/java.sql/share/classes/java/sql/ResultSetMetaData.java

```diff
@@ -180,7 +181,7 @@
      * @return precision
      * @exception SQLException if a database access error occurs
      */
-    int getPrecision(int column) throws SQLException;
+    @NonNegative int getPrecision(@Positive int column) throws SQLException;
```

Comment:
The Javadoc explicitly says that the return value may be 0, on line 177 above.  Why did you annotate the return type as `@NonNegative`?